File: templates/sample.py
# vim: set ts=4 sts=0 sw=4 si fenc=utf-8 et:
# vim: set fdm=marker fmr={{{,}}} fdl=0 foldcolumn=4:
# Authors:     FL
# Maintainers: FL
# Copyright:   YYYY, HRDAG, GPL v2 or later
# =========================================
# Project-Name/parent-task/core-task/src/script.py

# ---- dependencies {{{
from pathlib import Path
from sys import stdout
import argparse
import logging
import pandas as pd
#}}}

# ---- support methods {{{
def initial_asserts():
    return 1 


def get_args():
    parser = argparse.ArgumentParser()
    parser.add_argument("--input", default=None)
    parser.add_argument("--output", default=None)
    args = parser.parse_args()
    assert Path(args.input).exists()
    return args


def get_logger(sname, file_name=None):
    logger = logging.getLogger(sname)
    logger.setLevel(logging.DEBUG)
    formatter = logging.Formatter("%(asctime)s - %(levelname)s " +
                                  "- %(message)s", datefmt='%Y-%m-%d %H:%M:%S')
    stream_handler = logging.StreamHandler(stdout)
    stream_handler.setFormatter(formatter)
    logger.addHandler(stream_handler)
    if file_name:
        file_handler = logging.FileHandler(file_name)
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)
    return logger


def final_asserts(df):
    return 1
#}}}

# ---- main {{{
if __name__ == '__main__':
    # setup logging
    logger = get_logger(__name__, "output/script-name.log")

    # arg handling
    args = get_args()
    input_f = args.input
    output_f = args.output

    # read data, initial verification
    logger.info("Loading data.")
    raw_df = pd.read_ext(input_f)
    initial_asserts(raw_df)
    
    # do stuff, more verification
    final_asserts(raw_df)

    # save data, final verification
    raw_df.to_parquet(output_f)
    
    logger.info("done.")
    
#}}}
# done.

File: templates/sample.R
# vim: set ts=4 sts=0 sw=4 si fenc=utf-8 et:
# vim: set fdm=marker fmr={{{,}}} fdl=0 foldcolumn=4:
# Authors:     FL
# Maintainers: FL
# Copyright:   YYYY, HRDAG, GPL v2 or later
# =========================================
# Project-Name/parent-task/core-task/src/script.R

# ---- dependencies {{{
library(pacman)
pacman::p_load(argparse, assertr, logger, tidyverse)
#}}}

# ---- support methods {{{
get_args <- function() {
    parser <- ArgumentParser()
    parser$add_argument("--input")
    parser$add_argument("--output")
    args <- parser$parse_args()
    args
}

set_log <- function(logname) {
    log_appender(appender_tee(logname))
}

initial_asserts <- function(data) {
    data %>%
        verify(nrow(.) > 100)
}
# }}}

# ---- main {{{
logname <- "output/core-task.log"
set_log(logname)

# arg handling
args <- get_args() 
input <- args$input
output <- args$output

# read data, initial verification
log_info("Loading data", logger="")
dat <- read.ext(input)
initial_asserts(dat)

# save data
write.parquet(dat, output)

log_info("done", logger="")
#}}}

# done.
